### PR TITLE
Ft my places

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Enables search on my places list
- Replaces `Favourites` with `My Places`
- Disables displace of nodes that don't have the `last updated` value
- Fixes bugs caused by having different variable names pointing to the same functionality
- Adds secure storage of API keys

#### Is this change ready to hit production in its current state?
- [ ] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
Checkout  ``` ft-my-places ```
```
cd AirQo-frontend/mobile
```
Add [secret.ts](https://drive.google.com/file/d/15dukiSdmHaVk1du5mYAJ2rTHvDyy6DMV/view?usp=sharing) file to `src/providers/api` folder
Follow instructions in `Readme.md` on how to setup your environment and run the application

#### What are the relevant tickets?
- [PLAT-503](https://airqoteam.atlassian.net/browse/PLAT-503?atlOrigin=eyJpIjoiYmU3MWY4ZmIzMzdjNDE4MTkxMmNjNDNjN2JlZDRkNDQiLCJwIjoiaiJ9)
- [PLAT-542](https://airqoteam.atlassian.net/browse/PLAT-542?atlOrigin=eyJpIjoiMmIwYmVjZTJmNmNiNDYwZjkxYjA3ZDUwMGI3YmUyNmMiLCJwIjoiaiJ9)
- [PLAT-539](https://airqoteam.atlassian.net/browse/PLAT-539?atlOrigin=eyJpIjoiMzUwYWM0ZTBjNTBiNGVhYWI1MTM3NzlkYzlmYWMxY2QiLCJwIjoiaiJ9)
- [PLAT-540](https://airqoteam.atlassian.net/browse/PLAT-540?atlOrigin=eyJpIjoiZmM0MDA1MDUxZDNlNDc1MWJlOGVhMDhkMDJkZDBjZjEiLCJwIjoiaiJ9)

#### Screenshots (optional)

### CODE REVIEW

**Review by danielogen — COMMENTED** [review]
Tried testing this but am getting the error attached. I ran the following command after installing Cordova and Ionic globally: `npm install` and `ionic serve`.
Screenshot 2021-05-20 at 07 25 37

**Comment by buzentech** [comment]
> 
> 
> Tried testing this but am getting the error attached. I ran the following command after installing Cordova and Ionic globally: `npm install` and `ionic serve`.
> Screenshot 2021-05-20 at 07 25 37

Hello Daniel, 

Kindly reach out to Noah to provide you with that file.

**Comment by NoahNsimbe** [comment]
> Tried testing this but am getting the error attached. I ran the following command after installing Cordova and Ionic globally: `npm install` and `ionic serve`.
> Screenshot 2021-05-20 at 07 25 37

@danielogen, please revisit the PR description, there are some instructions on how to get and where to put the `secret.ts` file. Going to reshare via slack

**Comment by danielogen** [comment]
> > Tried testing this but am getting the error attached. I ran the following command after installing Cordova and Ionic globally: `npm install` and `ionic serve`.
> > Screenshot 2021-05-20 at 07 25 37
> 
> Hello Daniel,
> 
> Kindly reach out to Noah to provide you with that file.

Thanks all good now!

**Comment by danielogen** [comment]
> > Tried testing this but am getting the error attached. I ran the following command after installing Cordova and Ionic globally: `npm install` and `ionic serve`.
> > Screenshot 2021-05-20 at 07 25 37
> 
> @danielogen, please revisit the PR description, there are some instructions on how to get and where to put the `secret.ts` file. Going to reshare via slack

Thanks @NoahNsimbe , its working well now

**Review by danielogen — CHANGES_REQUESTED** [review]
Working well, just one more change to make; "Saved Places" -> "My Places". See screenshot
Screenshot 2021-05-20 at 09 20 29

**Comment by NoahNsimbe** [comment]
> Working well, just one more change to make; "Saved Places" -> "My Places". See screenshot
> Screenshot 2021-05-20 at 09 20 29

Thanks @danielogen

**Review by Baalmart — APPROVED** [review]
LGTM to me, all notes in the PR description are present.

File: mobile/src/pages/node/node.ts [hunk 1]
```diff
@@ -106,55 +125,40 @@ export class NodePage {
...
           this.offlineStoreHistoryStoreNodeInfo(result.feed.hourly_results, this.node);
           this.getHistoryGraphData(result.feed.hourly_results);
         }
         loader.dismiss();
       } else {
+        loader.dismiss();
 
         this.toastCtrl.create({
-          message: 'History information not available',
+          message: 'Up-to-date history information unavailable',
```
Comment (Baalmart) [inline mobile/src/pages/node/node.ts:147]: Just curious about how we can minimise this situation of not having up to date historical values. cc: @sserurich

**Review by danielogen — APPROVED** [review]
Hi @NoahNsimbe , things are working very well on my end. Just a quick inquiry: I see "My Location" and "My Places" -  I assume "My Location" is automatically inferred?

**Comment by danielogen** [comment]
> Hi @NoahNsimbe , things are working very well on my end. Just a quick inquiry: I see "My Location" and "My Places" - I assume "My Location" is automatically inferred?

Screen Shot 2021-06-17 at 11 22 52 AM

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**